python/private/py_wheel.bzl

```diff
@@ -17,6 +17,62 @@
 load("//python/private:stamp.bzl", "is_stamping_enabled")
 load(":py_package.bzl", "py_package_lib")
 
+RequirementsInfo = provider(
+    "Information about all the requirements of a py_wheel.",
+    fields = {
+        "requirements": "The list of requirements.",
+    },
+)
+
+def _requirements_collector_impl(_, ctx):
+    """Aspect to collect third-party dependencies.
+
+    Args:
+        _: The bazel target the aspect is applied to.
+        ctx: The aspect context.
+    Returns: RequirementsInfo
+    """
+
+    requirement = None
+    if hasattr(ctx.rule.attr, "tags"):
+        pypi_name = None
+        pypi_version = None
+        for tag in ctx.rule.attr.tags:
+            if tag.startswith("pypi_name="):
+                pypi_name = tag[len("pypi_name="):]
+            elif tag.startswith("pypi_version="):
+                pypi_version = tag[len("pypi_version="):]
+
+        if pypi_name and pypi_version:
+            requirement = "{}=={}".format(pypi_name, pypi_version)
+
+    if hasattr(ctx.rule.attr, "deps"):
```

Comment:
Similar comment to before: Add a comment explaining what is being visited where it's expected that it the rule won't have the attribute.

Same comment on the "srcs" check below, too

Reply:
When traversing the dependency tree, srcs and deps are not attributes of every rule implementation. Nothing fancy going on here.